.github/workflows/ci.yaml

```diff
@@ -0,0 +1,296 @@
+        name: Start CockroachDB
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ^1.16
+      - uses: actions/cache@v2
+        with:
+          path: ./test/e2e/hydra
+          key: ${{ runner.os }}-hydra
+      - run: ./test/e2e/circle-ci.bash ${{ matrix.database }} ${{ matrix.args }}
+
+  docs-build:
+    runs-on: ubuntu-latest
+    name: Build and publish docs
+    if: ${{ github.ref_type == 'tag' || github.ref_name == 'master' }}
+    needs:
+      - test
+    steps:
+      - uses: ory/ci/docs/build@master
+        with:
+          token: ${{ secrets.ORY_BOT_PAT }}
+          swag-spec-location: spec/api.json
+
+  docs-cli:
+    runs-on: ubuntu-latest
+    name: Build CLI docs
+    needs:
+      - test
+    steps:
+      - uses: ory/ci/docs/cli@master
+        with:
+          token: ${{ secrets.ORY_BOT_PAT }}
+
+  changelog:
+    name: Generate changelog
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' || github.ref_name == 'master' }}
+    needs:
+      - test
+      - test-hsm
+      - test-e2e
+      - docs-build
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: ory/ci/changelog@master
+        with:
+          token: ${{ secrets.ORY_BOT_PAT }}
+
+  sdk-generate:
+    name: Generate SDKs
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' || github.ref_name != 'master' || github.event_name == 'pull_request' }}
+    needs:
+      - test
+      - test-hsm
+      - test-e2e
+    steps:
+      - uses: ory/ci/sdk/generate@master
+        with:
+          token: ${{ secrets.ORY_BOT_PAT }}
+          app-name: Ory_Hydra
+          swag-spec-ignore: internal/httpclient gopkg.in/square/go-jose.v2
+          swag-spec-location: spec/api.json
+
+  sdk-release:
+    name: Release SDKs
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    needs:
+      - test
+      - test-hsm
+      - sdk-generate
+      - docs-build
+      - release
+    steps:
+      - uses: ory/ci/sdk/release@master
+        with:
+          token: ${{ secrets.ORY_BOT_PAT }}
+          app-name: Ory_Hydra
+          swag-spec-ignore: internal/httpclient gopkg.in/square/go-jose.v2
+          swag-spec-location: spec/api.json
+
+  release:
+    name: Generate release
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    needs:
+      - oidc-conformity
+      - test
+      - test-hsm
+      - test-e2e
+      - changelog
+      - docs-build
+    steps:
+      - uses: ory/ci/releaser@master
+        with:
+          token: ${{ secrets.ORY_BOT_PAT }}
+          goreleaser_key: ${{ secrets.GORELEASER_KEY }}
+          cosign_pwd: ${{ secrets.COSIGN_PWD }}
+          docker_username: ${{ secrets.DOCKERHUB_USERNAME }}
+          docker_password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+  render-version-schema:
+    name: Render version schema
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
```

Comment:
This should run on both branch (master / pull request) and tag. However, it should only push if the ref_type is a tag